### ISSUE TITLE
Add a dialog to download the Android SDK from the editor

### ISSUE THREAD

**Issue by ywmaa** [godotengine/godot-proposals#4351]
### Describe the project you are working on

Any android game

### Describe the problem or limitation you are having in your project

currently any android developer will need to download and setup android studio then chooses the sdk and ndk paths in Godot Engine.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add an option in Godot to auto download android studio ,sdk and ndk and setup the paths automatically.

overall this feature is just a feature that has for me a low priority, but maybe it could enhance Godot Export Setup.

also I am not sure if it is possible and if it will be against Godot MIT license.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

N/A

### If this enhancement will not be used often, can it be worked around with a few lines of script?

it will simplify a process for developers.
I am not sure if it will be available through lines of script.
but overall it can get ignored from the engine as we can just install android studio ourselves.

### Is there a reason why this should be core and not an add-on in the asset library?

N/A

**Comment by boruok** [comment]
why? just follow godot documentation. Like other engines do.

**Comment by ywmaa** [comment]
@boruok 
yeah it feels so simple, and I wouldn't actually mind doing this, but since you mentioned other engines, now unreal engine 5 downloads and installs android stuff through automated process (I think it is called "turnkey").

like I said I don't consider this as a must to have, but it is nice to have.

**Comment by ywmaa** [comment]
I am starting to think that this feature isn't good even if it is implemented in the Engine, I feel like it is specific to a platform, and extra code for relatively simple process.

if the docs are just updated when a new SDK version releases I think this feature is useless, and it will require keeping track of SDK releases,

should I close the issue ? 
or keep it opened to see how others think about it ?

**Comment by ywmaa** [comment]
also it feels like it will add bloat code.

**Comment by hani09876** [comment]
> also it feels like it will add bloat code.

I think. As you said it will bloat of code. Then rather than doing it on code can it be implement as a asset? By the way i also think this feature must have to be in godot. Because android setup for godot is too much problematic and irritating.

**Comment by ywmaa** [comment]
> > also it feels like it will add bloat code.
> 
> I think. As you said it will bloat of code. Then rather than doing it on code can it be implement as a asset? By the way i also think this feature must have to be in godot. Because android setup for godot is too much problematic and irritating. 

I think it can be added as a plugin yeah,

Overall android platform setup is annoying in almost all game engines (didn't try unity) 
But in unreal it is also annoying (didn't fully try the new unreal engine 5 turnkey)

But most issues come from docs being outdated and not following the current android studio version, or not being specific 
,sometimes compatibility breaks in play store because of how google decides to accept apps/games

For example in unreal I faced a problem where I can't have the game bigger than 1GB in install time because of play store restrictions and then if I want more size I must use chunks and divide it to 512MB each at a maximum of 4 chunks as I remember(google play app delivery), these are stuff at install time (which gives us total of 3GB install time ) 
Also I didn't mention how currently unreal engine doesn't automatically add the 512MB chunks to the aab and you must move it to a specific folder to work OMG it is very annoying 

But most of it is due to play store changing the way publishing works 
And the community in unreal engine isn't really very helpful and big about android games.

I didn't try big games in Godot yet, but I tried after installing the SDK and jre (linux repositories for jre) to export to android 
And it is very smooth and easy experience (and I actually tried it in Godot 4 alpha) , I think this is due to good docs, overall I am currently pleased with how Godot solved this problem, I hope I don't face more problems when the game becomes big.

Sorry for talking a lot, I wanted to share my experience.

Moral of the story : Good docs and following the updates of the market means easier setup and export

I think having these stuff directly in the engine will be problematic, let's say google decides to suddenly change something in the android setup, boom your script in Godot now isn't working( people will go to install it using Godot automated process, boom not working , then blame Godot for Play store problems) and it will put pressure on engine developers to maintain these stupid breaking updates.

**Comment by hani09876** [comment]
> > > also it feels like it will add bloat code.
> > 
> > I think. As you said it will bloat of code. Then rather than doing it on code can it be implement as a asset? By the way i also think this feature must have to be in godot. Because android setup for godot is too much problematic and irritating. 
> 
> I think it can be added as a plugin yeah,
> 
> Overall android platform setup is annoying in almost all game engines (didn't try unity) 
> But in unreal it is also annoying (didn't fully try the new unreal engine 5 turnkey)
> 
> But most issues come from docs being outdated and not following the current android studio version, or not being specific 
> ,sometimes compatibility breaks in play store because of how google decides to accept apps/games
> 
> For example in unreal I faced a problem where I can't have the game bigger than 1GB in install time because of play store restrictions and then if I want more size I must use chunks and divide it to 512MB each at a maximum of 4 chunks as I remember(google play app delivery), these are stuff at install time (which gives us total of 3GB install time ) 
> Also I didn't mention how currently unreal engine doesn't automatically add the 512MB chunks to the aab and you must move it to a specific folder to work OMG it is very annoying 
> 
> But most of it is due to play store changing the way publishing works 
> And the community in unreal engine isn't really very helpful and big about android games.
> 
> I didn't try big games in Godot yet, but I tried after installing the SDK and JDK (linux repositories for JDK) to export to android 
> And it is very smooth and easy experience (and I actually tried it in Godot 4 alpha) , I think this is due to good docs, overall I am currently pleased with how Godot solved this problem, I hope I don't face more problems when the game becomes big.
> 
> Sorry for talking a lot, I wanted to share my experience.
> 
> Moral of the story : Good docs and following the updates of the market means easier setup and export
> 
> I think having these stuff directly in the engine will be problematic, let's say google decides to suddenly change something in the android setup, boom your script in Godot now isn't working( people will go to install it using Godot automated process, boom not working , then blame Godot for Play store problems) and it will put pressure on engine developers to maintain these stupid breaking updates.

Yeah. Thats correct. Then what you decided will you make this as a plugin. I am facing two new problem. 
1. I can't check is the user is connected to the internet. 
2. I change the pitch scale of my audio for this my audio speed become faster. But i want to make set my audio speed as it is and change the pitch scale. 

There are two many user from 2018 (from my search on Google) who are searching these solution. But they didn't got any answer. I think these feature needs to be in core.

**Comment by ywmaa** [comment]
@hani09876 
No I don't think I am going to work on this plugin (atleast for now), currently I'm already working on a template, 
And there are other plugins in my mind that are worth my attention more than this.

1-
Check this out for internet connection :
https://godotengine.org/qa/16854/check-if-user-has-a-internet-connection-on-mobile-device

https://github.com/cagdasc/godot-network-checker

If you see it as a must to have feature in Godot open a proposal

2- I am not sure if this a feature or a bug 
If it is a bug open it in the main Godot repo.

**Comment by Calinou** [comment]
This was implemented by https://github.com/godotengine/godot/pull/121849, closing.
